cmd/update-main.go

```diff
@@ -462,14 +463,31 @@ func doUpdate(sha256Hex string, latestReleaseTime time.Time, ok bool) (updateSta
 		opts.Verifier = v
 	}
 
+	if e := opts.CheckPermissions(); e != nil {
+		permErrMsg := fmt.Sprintf(" failed with: %s", e)
+		updateStatusMsg = colorYellowBold("mc update to version RELEASE.%s %s.",
+			fmtReleaseTime, permErrMsg)
+		return updateStatusMsg, nil
+	}
+
 	if e = selfupdate.Apply(rc, opts); e != nil {
 		if re := selfupdate.RollbackError(e); re != nil {
+			rollBackErr := fmt.Sprintf("Failed to rollback from bad update: %v", re)
+			updateStatusMsg = colorYellowBold("mc update to version RELEASE.%s %s.", fmtReleaseTime, rollBackErr)
 			return updateStatusMsg, probe.NewError(e)
 		}
+
+		var pathErr *os.PathError
+		if errors.As(e, &pathErr) {
+			pathErrMsg := fmt.Sprintf("Unable to update the binary at %s: %v", filepath.Dir(pathErr.Path), pathErr.Err)
+			updateStatusMsg = colorYellowBold("mc update to version RELEASE.%s %s.",
+				fmtReleaseTime, pathErrMsg)
+			return updateStatusMsg, nil
+		}
 	}
```

Comment:
Missing return before exit ?

Reply:
I have updated. thank you so much for the clarification and guidance.